web/src/api/storage/types/checks.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import * as config from "./config";
+
+// Type guards.
+
+export function isFormattedDrive(drive: config.DriveElement): drive is config.FormattedDrive {
+  return "filesystem" in drive;
+}
+
+export function isPartitionedDrive(drive: config.DriveElement): drive is config.PartitionedDrive {
+  return !("filesystem" in drive);
+}
+
+export function isSearchAll(search: config.Search): search is config.SearchAll {
+  return search === "*";
+}
+
+export function isSearchByName(search: config.Search): search is config.SearchByName {
```

Comment:
Same than above, there are several ways to search for name. This only checks if the shortcut has been used and that's not obvious from the function name.

Reply:
Again same than above, I know it corresponds to the name of the type. But still I see to much room for confusion.